src/Plugin_TEST.cc

```diff
@@ -134,3 +135,229 @@ TEST(PluginTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(Getters))
+  ASSERT_NE(nullptr, ignGuiElem);
+
+  // Iterate properties
+  for (auto propElem = ignGuiElem->FirstChildElement("property");
+      propElem != nullptr;)
+  {
+    // If property in map, mark it as "Verified"
+    if (pluginProps.find(propElem->Attribute("key")) != pluginProps.end())
+    {
+      EXPECT_EQ(propElem->GetText(), pluginProps[propElem->Attribute("key")]);
+      pluginProps[propElem->Attribute("key")] = "Verified";
+    }
+    auto nextProp = propElem->NextSiblingElement("property");
+    propElem = nextProp;
+  }
+
+  // Verify all inputs properties are correct
+  for (auto itr = pluginProps.begin(); itr != pluginProps.end(); itr++)
+  {
+    EXPECT_EQ(itr->second, "Verified");
+  }
+
+}
+
+/////////////////////////////////////////////////
+TEST(PluginTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(ConfigStrInputNoPlugin))
+{
+  ignition::common::Console::SetVerbosity(4);
+
+  Application app(g_argc, g_argv);
+  app.AddPluginPath(std::string(PROJECT_BINARY_PATH) + "/lib");
+
+  // Load normal plugin
+  const char *pluginStr = "";
+
+  tinyxml2::XMLDocument pluginDoc;
+  pluginDoc.Parse(pluginStr);
+  EXPECT_TRUE(app.LoadPlugin("WorldStats",
+      pluginDoc.FirstChildElement("plugin")));
+
+  // Create main window
+  auto win = app.findChild<MainWindow *>();
+  ASSERT_NE(nullptr, win);
+
+  // Check plugin count
+  EXPECT_EQ(1, win->findChildren<Plugin *>().size());
+
+  // Get the output for ConfigStr()
+  std::string configStr;
+  tinyxml2::XMLDocument configDoc;
+  auto plugin = win->findChildren<Plugin *>()[0];
+  configStr = plugin->ConfigStr();
+  configDoc.Parse(configStr.c_str());
+
+  // ConfigStr() creates a plugin with default value when input doesn't
+  // contain a <plugin> tag.
+  // We select a few to verify.
+  std::unordered_map<std::string, std::string> pluginProps;
+  pluginProps["showTitleBar"] = "true";
+  pluginProps["resizable"] = "true";
+  pluginProps["height"] = "952";
+  pluginProps["width"] = "1199";
+  pluginProps["z"] = "0";
+  pluginProps["state"] = "docked";
+
+  // <plugin>
+  auto pluginElem = configDoc.FirstChildElement("plugin");
+  ASSERT_NE(nullptr, pluginElem);
+
+  // <ignition-gui>
+  auto ignGuiElem = pluginElem->FirstChildElement("ignition-gui");
+  ASSERT_NE(nullptr, ignGuiElem);
+
+  // Iterate properties
+  for (auto propElem = ignGuiElem->FirstChildElement("property");
+      propElem != nullptr;)
+  {
+    // If property in map, mark it as "Verified"
+    if (pluginProps.find(propElem->Attribute("key")) != pluginProps.end())
+    {
+      EXPECT_EQ(propElem->GetText(), pluginProps[propElem->Attribute("key")]);
+      pluginProps[propElem->Attribute("key")] = "Verified";
+    }
+    auto nextProp = propElem->NextSiblingElement("property");
+    propElem = nextProp;
+  }
+
+  // Verify all selected inputs properties are correct
+  for (auto itr = pluginProps.begin(); itr != pluginProps.end(); itr++)
+  {
+    EXPECT_EQ(itr->second, "Verified");
```

Comment:
In case this fails, it would be good to print out which one, eg.:
```suggestion
    EXPECT_EQ(itr->second, "Verified") << "Did not find property: " <<  itr->first;
```